library/core/src/num/mod.rs

```diff
@@ -1658,6 +1658,38 @@ mod verify {
         }
     }
 
+    // Part 2 : Nested unsafe functions Generation Macros --> https://github.com/verify-rust-std/blob/main/doc/src/challenges/0011-floats-ints.md
+
+    // Verify `widening_mul`, which internally uses `unchecked_mul`
+    macro_rules! generate_widening_mul_intervals {
+        ($type:ty, $wide_type:ty, $($harness_name:ident, $min:expr, $max:expr),+) => {
+            $(
+                #[kani::proof]
+                pub fn $harness_name() {
+                    let lhs: $type = kani::any::<$type>();
+                    let rhs: $type = kani::any::<$type>();
+
+                    kani::assume(lhs >= $min && lhs <= $max);
+                    kani::assume(rhs >= $min && rhs <= $max);
+
+                    // Ensure that multiplication in the wider type does not overflow
+                    kani::assume(!((lhs as $wide_type).overflowing_mul(rhs as $wide_type).1));
```

Comment:
Thanks for pointing that @celinval , upon checking the function, we already have the overflowing precondition for the unchecked_mul function : 
```
#[requires(!self.overflowing_mul(rhs).1)]
pub const unsafe fn unchecked_mul(self, rhs: Self) -> Self 
```

And the widening_mul calls the unchecked_mul, Hence we can safely remove this condition as its already being checked at function level. Updated code.

Reply:
Sorry, I forgot to mark the other assumptions above. Ideally, the harness shouldn't have any assumptions. Unless you bump into some issue like performance.

Reply:
Yes @celinval , The assumptions are added because of the performance constraint in multiplication for u16, u32 and u64 types. u8 shouldn't have an issue since the total number of checks is around 256 × 256 = 65,536.

But for u16, its around 65,536 × 65,536 ~ 4.29 billion, which is infeasible, hence have added the assumptions.

Reply:
Sorry... I totally missed that!